src/numerics/include/metaphysicl/dualnumber_surrogate.h

```diff
@@ -0,0 +1,316 @@
+#ifndef METAPHYSICL_DUALNUMBER_SURROGATE_H
+#define METAPHYSICL_DUALNUMBER_SURROGATE_H
+
+#include "metaphysicl/dualnumber_surrogate_decl.h"
+#include "metaphysicl/dualnumber_decl.h"
+
+namespace MetaPhysicL
+{
+
+template <typename T, std::size_t N>
+inline DualNumberSurrogate<T, N>::DualNumberSurrogate(DualNumber<T, NumberArray<N, T>> & dn)
+  : value(dn.value())
+{
+  for (decltype(N) i = 0; i < N; ++i)
+    derivatives[i] = &dn.derivatives()[i];
```

Comment:
In these cases where there are no args to forward, derivatives just ends up being the same as dn.derivatives, so do you need the manual for loop here?  Couldn't you even just construct derivatives in the member initializer list?

Reply:
`DualNumberSurrogate<T,N>::derivatives` is a `NumberArray<N, T*>`, not a `NumberArray<N,T> *`, e.g. I need to assign the addresses element-wise. Is there a smarter way to do this? I could be being dense

Reply:
Oh, no, that was me being dense.  Sorry, I should have edited my comment after I realized what you were doing.